File: SchoolWebApp-API/Project.Core/Entities/Class/LearningLevel.cs
using SchoolWebApp.Core.Entities.CBE.Assessments;
using SchoolWebApp.Core.Entities.School;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.Class
{
    public class LearningLevel : Base
    {
        public required string Name { get; set; }
        public int Rank { get; set; }
        public string? Description { get; set; }
        public int EducationLevelId { get; set; }
        public EducationLevel? EducationLevel { get; set; }

        public List<SchoolClass> SchoolClasses { get; set; } = new();
        public List<SpecificOutcome> SpecificOutcomes { get; set; } = new();
    }
}

File: SchoolWebApp-Client/src/app/class/models/learning-level.ts
import { ResourceModel } from "@/core/models/ResourceModel";
import { EducationLevel } from "@/school/models/educationLevel";

export class LearningLevel extends ResourceModel<LearningLevel> {
    public name?: string;
    public rank?: number;
    public description?: string;
    public educationLevelId?: number;
    public educationLevel?: EducationLevel;

    constructor(model?: Partial<LearningLevel>) {
        super(model);
      }
}